core/templates/domain/skill/concept-card-backend-api.service.ts

```diff
@@ -17,92 +17,116 @@
+import cloneDeep from 'lodash/cloneDeep';
 
-    var _fetchConceptCards = function(
-        skillIds, successCallback, errorCallback) {
-      var conceptCardDataUrl = UrlInterpolationService.interpolateUrl(
-        CONCEPT_CARD_DATA_URL_TEMPLATE, {
-          comma_separated_skill_ids: skillIds.join(',')
-        });
+import { ConceptCard, IConceptCardBackendDict, ConceptCardObjectFactory} from
+  'domain/skill/ConceptCardObjectFactory';
+import { SkillDomainConstants } from
+  'domain/skill/skill-domain.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+interface IConceptCardBackendDicts {
+  'concept_card_dicts': IConceptCardBackendDict[];
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConceptCardBackendApiService {
+  constructor(
+    private conceptCardObjectFactory: ConceptCardObjectFactory,
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
+
+  // Maps previously loaded concept cards to their IDs.
+  private _conceptCardCache = [];
 
-      $http.get(conceptCardDataUrl).then(function(response) {
-        var conceptCards = angular.copy(response.data.concept_card_dicts);
+  private _fetchConceptCards(
+      skillIds: string[],
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: string) => void): void {
+    var conceptCardDataUrl = this.urlInterpolation.interpolateUrl(
+      SkillDomainConstants.CONCEPT_CARD_DATA_URL_TEMPLATE, {
+        comma_separated_skill_ids: skillIds.join(',')
+      });
+
+    var conceptCardObjects = [];
+
+    this.http.get<IConceptCardBackendDicts>(conceptCardDataUrl).toPromise()
+      .then(response => {
         if (successCallback) {
-          successCallback(conceptCards);
+          var conceptCardDicts = response.concept_card_dicts;
+          conceptCardDicts.forEach((conceptCardDict) => {
+            conceptCardObjects.push(
+              this.conceptCardObjectFactory.createFromBackendDict(
+                conceptCardDict));
+          });
+          successCallback(conceptCardObjects);
         }
-      }, function(errorResponse) {
+      }, errorResponse => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(errorResponse.error);
         }
       });
-    };
+  }
 
-    var _isCached = function(skillId) {
-      return _conceptCardCache.hasOwnProperty(skillId);
-    };
+  private _isCached(skillId: string): boolean {
+    return this._conceptCardCache.hasOwnProperty(skillId);
+  }
 
-    var _getUncachedSkillIds = function(skillIds) {
-      var uncachedSkillIds = [];
-      skillIds.forEach(function(skillId) {
-        if (!_isCached(skillId)) {
-          uncachedSkillIds.push(skillId);
-        }
-      });
-      return uncachedSkillIds;
-    };
+  private _getUncachedSkillIds(skillIds: string[]): string[] {
+    var uncachedSkillIds = [];
+    skillIds.forEach((skillId) => {
```

Comment:
```suggestion
    skillIds.forEach(skillId => {
```
Ditto in other places in this file.

Reply:
Fixed!